vizro-core/src/vizro/actions/_actions_utils.py

```diff
@@ -66,34 +63,97 @@ def _apply_filters(
     return data_frame
 
 
-def _apply_filter_interaction(
-    data_frame: pd.DataFrame,
-    ctds_filter_interaction: List[CallbackTriggerDict],
-    target: str,
+def _apply_chart_filter_interaction(
+    data_frame: pd.DataFrame, target: str, ctd_filter_interaction: Dict[str, CallbackTriggerDict]
 ) -> pd.DataFrame:
-    for ctd in ctds_filter_interaction:
-        if not ctd["value"]:
+    ctd_click_data = ctd_filter_interaction["clickData"]
+    if not ctd_click_data["value"]:
+        return data_frame
+
+    source_chart_id: ModelID = ctd_click_data["id"]
+    source_chart_actions = _get_component_actions(model_manager[source_chart_id])
+    try:
+        custom_data_columns = model_manager[source_chart_id]["custom_data"]  # type: ignore[index]
+    except KeyError as exc:
+        raise KeyError(f"No `custom_data` argument found for source chart with id {source_chart_id}.") from exc
+
+    customdata = ctd_click_data["value"]["points"][0]["customdata"]
+
+    for action in source_chart_actions:
+        if target not in action.function["targets"]:
             continue
+        for custom_data_idx, column in enumerate(custom_data_columns):
+            data_frame = data_frame[data_frame[column].isin([customdata[custom_data_idx]])]
 
-        source_chart_id = ctd["id"]
-        source_chart_actions = _get_component_actions(model_manager[source_chart_id])
+    return data_frame
 
-        try:
-            custom_data_columns = model_manager[source_chart_id]["custom_data"]  # type: ignore[index]
-        except KeyError as exc:
-            raise KeyError(f"No `custom_data` argument found for source chart with id {source_chart_id}.") from exc
 
-        customdata = ctd["value"]["points"][0]["customdata"]
+def _get_parent_vizro_table(_underlying_table_id: str) -> Table:
+    from vizro.models import Table
 
-        for action in source_chart_actions:
-            if target not in action.function["targets"]:
-                continue
+    for _, table in model_manager._items_with_type(Table):
+        if table._underlying_table_id == _underlying_table_id:
+            return table
+    raise KeyError(f"No parent Vizro.Table component found for underlying table with id {_underlying_table_id}.")
```

Comment:
Again, could we make this more generic? So we don't reference the table specifically but look for models with an underlying `_figure_id`